types.go

```diff
@@ -55,6 +55,11 @@ type (
 		// Checksum ensures the integrity of packages
 		// refereced by URL. Ignored for literals.
 		Checksum Checksum `json:"checksum"`
+	}
+
+	PackageRef struct {
+		PackageName      string
+		PackageNamespace string
```

Comment:
The idea is to separate packages and functions into two primitives: a low level "dumb" primitive that knows nothing about versions and a higher level thing that operates as a "factory" for packages or functions. Kinda like how Deployments act as factories for Pods.